bluesky/plans.py

```diff
@@ -2352,10 +2376,11 @@ def spiral_fermat(detectors, x_motor, y_motor, x_start, y_start, x_range,
         md = {}
     md = ChainMap(
         md,
-        {'detectors': [detector.name for detector in detectors],
-         'motors': [motor.name for motor in [x_motor, y_motor]],
+        {# 'detectors': inserted by scan_nd
+         # 'motors': inserted by scan_nd
+         # 'num_steps': inserted by scan_nd
          'plan_args': {'detectors': list(map(repr, detectors)),
-                       'x_motor': x_motor, 'y_motor': y_motor,
+                       'x_motor': repr(x_motor), 'y_motor': repr(y_motor),
```

Comment:
:+1: The existing implementation was a mistake; this is more consistent and more useful.